### ISSUE TITLE
khal fails with CollectionNotFoundError on a freshly configured vdirsyncer folder

### ISSUE THREAD

**Issue by muelli** [pimutils/khal#991]
I've just played around with various CalDAV tools and came across khal.
I've configured vdirsyncer and it looks to me like it has successfully run.
I've then ran khal configure.

```
➜ ~>cat /home/muelli/.config/khal/config
[calendars]

[[muelli_local]]
path = /tmp/muelli-calendar/*.ics
type = discover

[locale]
timeformat = %H:%M
dateformat = %Y-%m-%d
longdateformat = %Y-%m-%d
datetimeformat = %Y-%m-%d %H:%M
longdatetimeformat = %Y-%m-%d %H:%M

➜ ~>/tmp/gks3/bin/khal --verbosity DEBUG printcalendars
debug: khal 0.10.2
debug: using the config file at /home/muelli/.config/khal/config
debug: discovering calendars in /tmp/muelli-calendar/*.ics
debug: Found no or empty file `color` in /tmp/muelli-calendar/aa0488b2-1edd-4868-9b82-6e73270f855f.ics
debug: using collection's color for /tmp/muelli-calendar/aa0488b2-1edd-4868-9b82-6e73270f855f.ics
Traceback (most recent call last):
  File "/tmp/gks3/bin/khal", line 8, in <module>
    sys.exit(main_khal())
  File "/tmp/gks3/lib/python3.8/site-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/tmp/gks3/lib/python3.8/site-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/tmp/gks3/lib/python3.8/site-packages/click/core.py", line 1024, in invoke
    Command.invoke(self, ctx)
  File "/tmp/gks3/lib/python3.8/site-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmp/gks3/lib/python3.8/site-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/tmp/gks3/lib/python3.8/site-packages/click/decorators.py", line 16, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/tmp/gks3/lib/python3.8/site-packages/khal/cli.py", line 244, in cli
    prepare_context(ctx, config)
  File "/tmp/gks3/lib/python3.8/site-packages/khal/cli.py", line 201, in prepare_context
    conf = get_config(config)
  File "/tmp/gks3/lib/python3.8/site-packages/khal/settings/settings.py", line 137, in get_config
    config_checks(user_config, _get_color_from_vdir, _get_vdir_type)
  File "/tmp/gks3/lib/python3.8/site-packages/khal/settings/utils.py", line 239, in config_checks
    name = get_unique_name(vdir, config['calendars'].keys())
  File "/tmp/gks3/lib/python3.8/site-packages/khal/settings/utils.py", line 170, in get_unique_name
    name = Vdir(path, '.ics').get_meta('displayname')
  File "/tmp/gks3/lib/python3.8/site-packages/khal/khalendar/vdir.py", line 151, in __init__
    raise CollectionNotFoundError(path)
khal.khalendar.vdir.CollectionNotFoundError: /tmp/muelli-calendar/aa0488b2-1edd-4868-9b82-6e73270f855f.ics

➜ ~>ls -la /tmp/muelli-calendar/aa0488b2-1edd-4868-9b82-6e73270f855f.ics
-rw------- 1 muelli muelli 3533 Dez 19 16:31 /tmp/muelli-calendar/aa0488b2-1edd-4868-9b82-6e73270f855f.ics
➜ ~>
```

From looking at the code, it seems it expected a directory.
https://github.com/pimutils/khal/blob/4621f309dfb5945ee79665f8c3461b49dbcebcd8/khal/khalendar/vdir.py#L150

Because it mentions collections, I have configured vdirsyncer with `collections = null`:

```
[pair me]
a = "remote"
b = "muelli_local"
collections = null
```

I expected khal to do something useful rather than crash.

**Comment by d7415** [comment]
It is indeed expecting a directory. khal expects its calendars in vdir format. It could present a more useful error to the user though.

**Comment by muelli** [comment]
With `collections = null` in the configuration, vdirsyncer seems to put everything into the directory rather than create a directory for each collection.
It could be interesting to detect that case from `khal configure`.

**Comment by d7415** [comment]
> With `collections = null` in the configuration, vdirsyncer seems to put everything into the directory rather than create a directory for each collection.

From the vdirsyncer [docs](https://vdirsyncer.readthedocs.io/en/stable/tutorial.html#more-information-about-collections):
> null: The parameters give to the storage are exact and require no discovery.

Basically, you told it not to.

**Comment by muelli** [comment]
yeah, I guess I did something weird. But I didn't know what I was doing and rather just trying to make it work by copy and pasting examples I've found. I guess that when I didn't know what to put for "collections", because I didn't think that I was told how to obtain possible values, I looked into the documentation for the stanza https://vdirsyncer.readthedocs.io/en/stable/config.html#pair-section and found that `collections = null` there. And it seemed to work fine.
